pkg/image/pull_policy.go

```diff
@@ -1,41 +1,288 @@
+	// Pull images hourly
+	PullHourly
+	// Pull images daily
+	PullDaily
+	// Pull images weekly
+	PullWeekly
+	// Never pull images, even if they are not present
 	PullNever
 	// PullIfNotPresent pulls images if they aren't present
 	PullIfNotPresent
+	// PullWithInterval pulls images with specified intervals
+	PullWithInterval
 )
 
-var nameMap = map[string]PullPolicy{"always": PullAlways, "never": PullNever, "if-not-present": PullIfNotPresent, "": PullAlways}
+type Interval struct {
+	PullingInterval string `json:"pulling_interval"`
+	PruningInterval string `json:"pruning_interval"`
+	LastPrune       string `json:"last_prune"`
+}
+
+type ImageData struct {
+	ImageIDtoTIME map[string]string
+}
 
-// ParsePullPolicy from string
+type ImageJSON struct {
+	Interval *Interval  `json:"interval"`
+	Image    *ImageData `json:"image"`
+}
+
+var nameMap = map[string]PullPolicy{"always": PullAlways, "hourly": PullHourly, "daily": PullDaily, "weekly": PullWeekly, "never": PullNever, "if-not-present": PullIfNotPresent, "": PullAlways}
+
+// ParsePullPolicy from string with support for interval formats
 func ParsePullPolicy(policy string) (PullPolicy, error) {
 	if val, ok := nameMap[policy]; ok {
+		if val == PullHourly {
+			err := updateImageJSONDuration(hourly)
+			if err != nil {
+				return PullAlways, err
+			}
+		}
+		if val == PullDaily {
+			err := updateImageJSONDuration(daily)
+			if err != nil {
+				return PullAlways, err
+			}
+		}
+		if val == PullWeekly {
+			err := updateImageJSONDuration(weekly)
+			if err != nil {
+				return PullAlways, err
+			}
+		}
 		return val, nil
 	}
 
+	if strings.HasPrefix(policy, "interval=") {
+		interval = policy
+		intervalStr := strings.TrimPrefix(policy, "interval=")
+		matches := intervalRegex.FindStringSubmatch(intervalStr)
+		if len(matches) == 0 {
+			return PullAlways, errors.Errorf("invalid interval format: %s", intervalStr)
+		}
+
+		err := updateImageJSONDuration(intervalStr)
+		if err != nil {
+			return PullAlways, err
+		}
+
+		return PullWithInterval, nil
+	}
+
 	return PullAlways, errors.Errorf("invalid pull policy %s", policy)
 }
 
 func (p PullPolicy) String() string {
 	switch p {
 	case PullAlways:
 		return "always"
+	case PullHourly:
+		return "hourly"
+	case PullDaily:
+		return "daily"
+	case PullWeekly:
+		return "weekly"
 	case PullNever:
 		return "never"
 	case PullIfNotPresent:
 		return "if-not-present"
+	case PullWithInterval:
+		return fmt.Sprintf("%v", interval)
 	}
 
 	return ""
 }
+
+func updateImageJSONDuration(intervalStr string) error {
+	imageJSON, err := ReadImageJSON(logging.NewSimpleLogger(os.Stderr))
+	if err != nil {
+		return err
+	}
+
+	imageJSON.Interval.PullingInterval = intervalStr
+
+	updatedJSON, err := json.MarshalIndent(imageJSON, "", "    ")
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal updated records")
+	}
+
+	return WriteFile(updatedJSON)
+}
+
+func ReadImageJSON(l logging.Logger) (*ImageJSON, error) {
```

Comment:
What about creating a handler for these methods? I think this `ReadImageJSON` has a lot of responsibilities:
- It is trying to discover the user's home directory
- it creates the whole folder structure to save the `image.json` on the disk
- it reads and writes! 

```golang
type ImagePullPolicyManager struct {
	Logger logging.Logger
}

func (i *ImagePullPolicyManager) Read(path string) (*ImageJSON, error) {
  	imagePath := filepath.Join(path, "image.json")
	// Check if the file exists, if not, create it with minimum JSON
	if _, err := os.Stat(imagePath); os.IsNotExist(err) {
                // We don't need to write on disk, we are reading, right? 
                // We return our default 
		return &ImagePullPolicy{
			Interval: &Interval{
				PullingInterval: "7d",
				LastPrune:       "",
			},
			Image: &ImageData{},
		}, nil
	}
        // do what you are doing today
}

func (i *ImagePullPolicyManager) Write(imagePullPolicy ImagePullPolicy, path string) error {
	// I will move all the marshaling logic into these methods, I don't want the method caller to worry about them
        updatedJSON, err := json.MarshalIndent(imagePullPolicy, "", "    ")
	if err != nil {
		return errors.Wrap(err, "failed to marshal updated records")
	}
	return WriteFile(updatedJSON, filepath.Join(path, "image.json"))
}

func WriteFile(data []byte, path string) error {
        // Try this one, it will create the file if it doesn't exits.
	file, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0644)
	if err != nil {
		return errors.New("failed to open file: " + err.Error())
	}
	defer file.Close()
	_, err = file.Write(data)
	if err != nil {
		return errors.New("failed to write data to file: " + err.Error())
	}
	return nil
}

```

Reply:
Yes, this looks way better than what I was currently doing.